guides/source/upgrading_ruby_on_rails.md

```diff
@@ -276,6 +276,22 @@ class Notifier < ActionMailer::Base
 end
 ```
 
+### Foreign Key support
+
+Rails 4.2 adds basic support for foreign keys. If the
+[Foreigner gem](https://github.com/matthuhiggins/foreigner) were been used to
+get support for foreign keys in the app, you can consider removing the gem
+from Gemfile on upgrading to 4.2 and using the default support.
+
+But remember that some optional and adapter specific options might not work as
+expected, since Rails has just basic support. The naming convention for
+generating the constraint names is also different (the names are randomly
+generated), but this shouldn't cause problems.
+
+After removing Foreigner from Gemfile and running `bundle install`, it's
+important to run `rake db:schema:dump` and check if all foreign keys were
+dumped in `db/schema.rb`.
```

Comment:
would be nice to group the necessary steps to replace the gem in one place. Something along the lines of:

1. remove `gem "foreigner"` from the Gemfile
1. run `bundle install`
1. run `bin/rake db:schema:dump`
1. make sure that `db/schema.rb` contains every foreign key definition with the necessary options.